### PR TITLE
UCT/BASE: VFS nodes representing UCT components and memory domains.

### PR DESCRIPTION
**Pull request by rakhmets**
## What
Added VFS node representing UCT component.
Added VFS nodes representing UCT memory domain and its features.
Added symbolic link from UCT iface to UCT memory domain.
Added symbolic link from UCT memory domain to UCS registration cache.

### CODE REVIEW

**Review by yosefe — COMMENTED** [review]
LGTM, besides https://github.com/openucx/ucx/pull/7284#discussion_r697560536

File: src/uct/base/uct_md_vfs.c [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+    for (i = 0; i < ucs_static_array_size(uct_md_vfs_cap_infos); ++i) {
+        ucs_vfs_obj_add_ro_file(md, uct_md_vfs_read_cap,
+                                (void*)&uct_md_vfs_cap_infos[i].offset,
+                                uct_md_vfs_cap_infos[i].type, "capability/%s",
+                                uct_md_vfs_cap_infos[i].name);
+    }
+
+    ucs_vfs_obj_add_ro_file(md, uct_md_vfs_read_reg_cost, NULL, 0, "reg_cost");
+    ucs_vfs_obj_add_ro_file(md, uct_md_vfs_read_rkey_packed_size, NULL, 0,
```
[the remark this answers is not in the document]
Comment (rakhmets) [reply src/uct/base/uct_md_vfs.c]: Done

File: src/uct/base/uct_vfs_attr.c [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2001-2021.  ALL RIGHTS RESERVED.
```
Comment (brminich) [inline src/uct/base/uct_vfs_attr.c:2]: plz use `Copyright (C) 2021 NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.`, 
like in other new files

Comment (rakhmets) [reply src/uct/base/uct_vfs_attr.c]: Fixed

**Review by brminich — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Pull request merged**